services/management/backup/backups_service.go

```diff
@@ -0,0 +1,157 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package backup
+
+import (
+	"context"
+	"time"
+
+	"github.com/AlekSi/pointer"
+	"github.com/google/uuid"
+	backupv1beta1 "github.com/percona/pmm/api/managementpb/backup"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+//go:generate mockery -name=jobsService -case=snake -inpkg -testonly
+
+// jobsService is a subset of methods of agents.JobsService used by this package.
+// We use it instead of real type for testing and to avoid dependency cycle.
+type jobsService interface {
+	StopJob(jobID string) error
+	StartMySQLBackupRestoreJob(
+		jobID string,
+		pmmAgentID string,
+		serviceID string,
+		timeout time.Duration,
+		name string,
+		locationConfig models.BackupLocationConfig,
+	) error
+}
+
+// BackupsService represents backups API.
+type BackupsService struct {
+	db          *reform.DB
+	jobsService jobsService
+}
+
+// NewBackupsService creates new backups API service.
+func NewBackupsService(db *reform.DB, jobsService jobsService) *BackupsService {
+	return &BackupsService{
+		db:          db,
+		jobsService: jobsService,
+	}
+}
+
+// RestoreBackup starts restore backup job.
+func (s *BackupsService) RestoreBackup(
+	ctx context.Context,
+	req *backupv1beta1.RestoreBackupRequest,
+) (*backupv1beta1.RestoreBackupResponse, error) {
+	var restoreID string
+
+	var artifact *models.Artifact
+	var location *models.BackupLocation
+	var service *models.Service
+	var job *models.JobResult
+
+	err := s.db.InTransactionContext(ctx, nil, func(tx *reform.TX) error {
+		var err error
+		q := tx.Querier
+
+		service, err = models.FindServiceByID(q, req.ServiceId)
+		if err != nil {
+			return err
+		}
+
+		location, err = models.FindBackupLocationByID(q, req.LocationId)
+		if err != nil {
+			return err
+		}
+
+		artifact, err = models.FindArtifactByID(q, req.ArtifactId)
+		if err != nil {
+			return err
+		}
+
+		agents, err := models.FindPMMAgentsForService(tx.Querier, req.ServiceId)
+		if err != nil {
+			return err
+		}
+
+		if len(agents) != 1 {
```

Comment:
not sure about this, I'm checking if there is at least one agent, not exactly one.

Reply:
reworked